javascore/irc2Tradeable/src/main/java/foundation/icon/btp/irc2Tradeable/IRC2Tradeable.java

```diff
@@ -0,0 +1,146 @@
+/*
+ * Copyright 2021 ICON Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package foundation.icon.btp.irc2Tradeable;
+
+import foundation.icon.btp.lib.OwnerManager;
+import foundation.icon.btp.lib.OwnerManagerImpl;
+import score.*;
+import score.annotation.EventLog;
+import score.annotation.External;
+import score.annotation.Payable;
+import score.annotation.Optional;
+
+import java.math.BigInteger;
+import java.util.List;
+
+public class IRC2Tradeable extends IRC2Basic implements OwnerManager {
+    private final BranchDB<Address, DictDB<Address, BigInteger>> allowances = Context.newBranchDB("allowances", BigInteger.class);
+    private final Address ZERO_ADDRESS = Address.fromString("hx0000000000000000000000000000000000000000");
+
+    private final OwnerManager ownerManager = new OwnerManagerImpl("owners");
+
+    public IRC2Tradeable(String _name, String _symbol, int _decimals) {
+        super(_name, _symbol, _decimals);
+    }
+
+    @External
+    public void burn(BigInteger _amount) {
```

Comment:
Does the burn needs the onlyOwner, similar to the Solidity?

Reply:
You're right. This token should only burn by the Owner. Normal users should not allow burning their token

Reply:
This is all I have, Rest of the things looking good to me, Once you add the OnlyOwner, ill resolve this